### DISCUSSION TITLE
Proper installation and setup?

### DISCUSSION THREAD

**Discussion by AndnixSH** [vfsfitvnm/frida-il2cpp-bridge#111] [Q&A]
This is the first time compiling index.ts and i'm getting stuck, i'm really new to node.js.

Since README didn't explained the installation, i did research myself and found out i need to install node.js and tsc on Windows 10
Then i proceed with these steps as they were required

```
npm install --save-dev frida-il2cpp-bridge
npm install @types/node --save-dev
npm install @types/frida-gum --save-dev
tsc index.ts
```

But i'm getting duplicate error

```
C:/Users/andnix/AppData/Roaming/npm/node_modules/typescript/lib/lib.dom.d.ts:5465:11 - error TS2300: Duplicate identifier 'File'.

5465 interface File extends Blob {
               ~~~~

  node_modules/@types/frida-gum/index.d.ts:2200:15
    2200 declare class File {
                       ~~~~
    'File' was also declared here.

C:/Users/andnix/AppData/Roaming/npm/node_modules/typescript/lib/lib.dom.d.ts:5470:13 - error TS2300: Duplicate identifier 'File'.

5470 declare var File: {
                 ~~~~

  node_modules/@types/frida-gum/index.d.ts:2200:15
    2200 declare class File {
                       ~~~~
    'File' was also declared here.

node_modules/@types/frida-gum/index.d.ts:151:18 - error TS2300: Duplicate identifier 'gc'.

151 declare function gc(): void;
                     ~~

  node_modules/@types/node/globals.d.ts:55:13
    55 declare var gc: undefined | (() => void);
                   ~~
    'gc' was also declared here.

node_modules/@types/frida-gum/index.d.ts:2200:15 - error TS2300: Duplicate identifier 'File'.

2200 declare class File {
                   ~~~~

  C:/Users/andnix/AppData/Roaming/npm/node_modules/typescript/lib/lib.dom.d.ts:5465:11
    5465 interface File extends Blob {
                   ~~~~
    'File' was also declared here.
  C:/Users/andnix/AppData/Roaming/npm/node_modules/typescript/lib/lib.dom.d.ts:5470:13
    5470 declare var File: {
                     ~~~~
    and here.

node_modules/@types/node/globals.d.ts:55:13 - error TS2300: Duplicate identifier 'gc'.

55 declare var gc: undefined | (() => void);
               ~~

  node_modules/@types/frida-gum/index.d.ts:151:18
    151 declare function gc(): void;
                         ~~
    'gc' was also declared here.


Found 5 errors.
```

If i deleted the `node` folder which i installed via `npm install @types/node --save-dev` then i'm getting missing errors. Too long to post here

So what i'm missing there? is it improper way I installed it? If so, what's the proper way to install and setup?

**Comment by vfsfitvnm** [comment]
@AndnixSH 
There isn't any full-installation guide because this is a frida module. You must be familiar with frida and npm/ts project ecosystem.

You can read more about frida [here](https://frida.re/docs/javascript-api/).
Also, the `example` folder contains a minimal project sample (`index.ts`, `tsconfig.json`, `package.json`).

Since this isn't really related to my project, let me know if I can close this.

**Comment by AndnixSH** [comment]
Oh i was expecting more python related. Last question, can't this be used within python? I'm more familar with python and have frida installed
You can close it

**Comment by AndnixSH** [comment]
Nvm, frida is more related to Typescript.
And closed

**Comment by vfsfitvnm** [comment]
@AndnixSH 
Frida embeds a JS runtime (v8 or quickjs), so it can only run a single javascript file.

You can look at the example package.json file and look at the npm scripts; it uses frida-compile to bundle a multi module typescript project into a single javascript file.
Btw if you need a further explaination, there's a telegram group for frida: @fridadotre.
